src/api/useSearchVenuesOffer/useSearchVenueOffers.ts

```diff
@@ -0,0 +1,130 @@
+import { SearchResponse } from '@algolia/client-search'
+import { flatten } from 'lodash'
+import { useMemo, useRef } from 'react'
+import { useInfiniteQuery } from 'react-query'
+
+import { VenueListItem } from 'features/offer/components/VenueSelectionList/VenueSelectionList'
+import { useIsUserUnderage } from 'features/profile/helpers/useIsUserUnderage'
+import { initialSearchState } from 'features/search/context/reducer'
+import { formatFullAddress } from 'libs/address/useFormatFullAddress'
+import { AlgoliaHit } from 'libs/algolia'
+import { useSearchAnalyticsState } from 'libs/algolia/analytics/SearchAnalyticsWrapper'
+import { fetchOffers } from 'libs/algolia/fetchAlgolia/fetchOffers'
+import { useTransformOfferHits } from 'libs/algolia/fetchAlgolia/transformOfferHit'
+import { Position } from 'libs/geolocation'
+import { QueryKeys } from 'libs/queryKeys'
+import { getNextPageParam } from 'shared/getNextPageParam/getNextPageParam'
+import { Offer } from 'shared/offer/types'
+
+export type UseSearchVenueOffersType = {
+  offerId: number
+  query: string
+  geolocation: Position
+  shouldExecuteQuery: boolean
+  venueId?: number
+}
```

Comment:
Tu pourrais simplement exposer les paramètres de `useQuery`
```suggestion
export type UseSearchVenueOffersType = {
  offerId: number
  query: string
  geolocation: Position
  venueId?: number
  queryOptions?: Omit<QueryObserverOptions, "getNextPageParam">
}
```

Ca facilite l'utilisation, et ça te permet d'ajouter au besoin les paramètres nécessaires à l'avenir. Et en plus t'as pas besoin de gérer les types.